src/coreclr/jit/lclmorph.cpp

```diff
@@ -1125,11 +1192,17 @@ class LocalAddressVisitor final : public GenTreeVisitor<LocalAddressVisitor>
                 }
                 else if (indir->TypeIs(TYP_SIMD12))
                 {
-                    if ((offset == 0) && m_compiler->IsBaselineSimdIsaSupported())
+                    if ((offset == 0) && (varDsc->TypeGet() == TYP_SIMD16) && m_compiler->IsBaselineSimdIsaSupported())
```

Comment:
We could adjust this to handle `TYP_SIMD32`/`TYP_SIMD64` if desired, correct? Just not worth it as part of this PR?

Reply:
Yeah, let's leave that for a separate PR if desired. This was just a drive-by fix for #89456.